efficientdet/inference.py

```diff
@@ -213,6 +213,55 @@ def det_post_process(params: Dict[Any, Any],
     detections_batch: a batch of detection results. Each detection is a tensor
       with each row representing [image_id, x, y, width, height, score, class].
   """
+  batch_size = params['batch_size']
+  num_classes = params['num_classes']
+  # Create anchor_label for picking top-k predictions.
+  eval_anchors = anchors.Anchors(params['min_level'], params['max_level'],
+                                 params['num_scales'], params['aspect_ratios'],
+                                 params['anchor_scale'], params['image_size'])
+  anchor_labeler = anchors.AnchorLabeler(eval_anchors, num_classes)
+  if params['batch_size'] is None:
+    batch_size = tf.shape(list(cls_outputs.values())[0])[0]
+    cls_outputs_all = []
+    box_outputs_all = []
+    # Concatenates class and box of all levels into one tensor.
+    for level in range(params['min_level'], params['max_level'] + 1):
+      if params['data_format'] == 'channels_first':
+        cls_outputs[level] = tf.transpose(cls_outputs[level], [0, 2, 3, 1])
+        box_outputs[level] = tf.transpose(box_outputs[level], [0, 2, 3, 1])
+
+      cls_outputs_all.append(tf.reshape(
+          cls_outputs[level],
+          [batch_size, -1, num_classes]))
+      box_outputs_all.append(tf.reshape(
+          box_outputs[level], [batch_size, -1, 4]))
+    cls_outputs_all = tf.concat(cls_outputs_all, 1)
+    box_outputs_all = tf.concat(box_outputs_all, 1)
+    anchor_boxes = anchor_labeler._anchors.boxes
+    scores = tf.math.sigmoid(cls_outputs_all)
+    # apply bounding box regression to anchors
+    boxes = anchors.decode_box_outputs_tf(box_outputs_all, anchor_boxes)
+    boxes = tf.expand_dims(boxes, axis=2)
+    nmsed_boxes, nmsed_scores, nmsed_classes, valid_detections = tf.image.combined_non_max_suppression(
+      boxes, scores,
+      anchors.MAX_DETECTIONS_PER_IMAGE,
+      anchors.MAX_DETECTIONS_PER_IMAGE,
+      score_threshold=min_score_thresh,
+      clip_boxes=False)
+    height = nmsed_boxes[..., 2] - nmsed_boxes[..., 0]
+    width = nmsed_boxes[..., 3] - nmsed_boxes[..., 1]
+    detections = tf.stack([
```

Comment:
Why do we need a tf.tile here?  Could you add a simple comment?

Reply:
Actually, the whole image_id part can be just " image_ids = tf.ones([batch_size, max_boxes_to_draw], dtype=tf.float32)", since the image_ids are fake anyway for inference.

Reply:
I'm not sure whether developer will use image_id. If use ```tf.ones``` they could only get 1.